pkg/workload/resource_test.go

```diff
@@ -0,0 +1,169 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workload
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	nodev1 "k8s.io/api/node/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+func TestAdjustResources(t *testing.T) {
+	cases := map[string]struct {
+		runtimeClasses []nodev1.RuntimeClass
+		limitranges    []corev1.LimitRange
+		wl             *kueue.Workload
+		wantWl         *kueue.Workload
+	}{
+		"Handle Pod Overhead": {
+			runtimeClasses: []nodev1.RuntimeClass{
+				utiltesting.MakeRuntimeClass("runtime-a", "handler-a").
+					PodOverhead(corev1.ResourceList{
+						corev1.ResourceCPU:    ResourceQuantity(corev1.ResourceCPU, 1),
+						corev1.ResourceMemory: ResourceQuantity(corev1.ResourceMemory, 1024),
+					}).
+					RuntimeClass,
+			},
+			wl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						RuntimeClass("runtime-a").
+						Obj(),
+				).
+				Obj(),
+			wantWl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						RuntimeClass("runtime-a").
+						PodOverHead(
+							corev1.ResourceList{
+								corev1.ResourceCPU:    ResourceQuantity(corev1.ResourceCPU, 1),
+								corev1.ResourceMemory: ResourceQuantity(corev1.ResourceMemory, 1024),
+							}).
+						Obj(),
+				).
+				Obj(),
+		},
+		"Handle Pod Limit Range": {
+			limitranges: []corev1.LimitRange{
+				utiltesting.MakeLimitRange("foo", "").
+					WithType(corev1.LimitTypeContainer).
+					WithValue(
+						"Default", corev1.ResourceCPU, "4",
+					).
+					WithValue(
+						"DefaultRequest", corev1.ResourceCPU, "4",
+					).
+					WithValue(
+						"Max", corev1.ResourceCPU, "4",
+					).
+					WithValue(
+						"Min", corev1.ResourceCPU, "2",
+					).
+					LimitRange,
+			},
+			wl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						Obj(),
+					*utiltesting.MakePodSet("b", 1).
+						Limit(corev1.ResourceCPU, "5").
+						Obj(),
+					*utiltesting.MakePodSet("c", 1).
+						Request(corev1.ResourceCPU, "1").
+						Obj(),
+				).
+				Obj(),
+			wantWl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						Limit(corev1.ResourceCPU, "4").
+						Request(corev1.ResourceCPU, "4").
+						Obj(),
+					*utiltesting.MakePodSet("b", 1).
+						Limit(corev1.ResourceCPU, "5").
+						Request(corev1.ResourceCPU, "4").
+						Obj(),
+					*utiltesting.MakePodSet("c", 1).
+						Limit(corev1.ResourceCPU, "4").
+						Request(corev1.ResourceCPU, "1").
+						Obj(),
+				).
+				Obj(),
+		},
+		"Limits applied to requests": {
```

Comment:
Also take initContainers into consideration.

Reply:
Sorry,It's my mistake fail to recognize that part.I have taken initContainer into consideration now.